swift-tools-support-core/Sources/TSCTestSupport/PseudoTerminal.swift

```diff
@@ -31,10 +31,13 @@ public final class PseudoTerminal {
     }
     
     public func readMaster(maxChars n: Int = 1000) -> String? {
-        var buf: [CChar] = [CChar](repeating: 0, count: n)
-        if read(master, &buf, n) <= 0 {
-            return nil
+        let buf: [CChar] = .init(unsafeUninitializedCapacity: n) { buf, initializedCapacity in
```

Comment:
It would probably be good to create a buffer of `n + 1` instead of `n` here, and then read up to `n` instead of `n -1` bytes below.  It's a small nuance, and client's shouldn't count on receiving the maximum requested number of bytes anyway, but in this new implementation it would always receive at most one less than the requested maximum.

Reply:
Agree. This change makes sense. Fixed.